Improve comment of ExpectedValue in db stress

hx235 opened the pull request:
**Context/Summary:**
https://github.com/facebook/rocksdb/pull/11424 made me realize there are a couple gaps in my `ExpectedValue` comments so I updated them, along with separating `ExpectedValue` into separate files so it's clearer that `ExpectedValue` can be used without updating `ExpectedState` (e.g, TestMultiGet() where we care about value base of expected value but not updating the ExpectedState).


**Test:**
CI

ajkr commented:
Is it a common pattern to put declarations and definitions into differently named files? MustHaveNotExisted() would be in expected_state.cc and expected_value.h according to this PR. I know we split db_impl.h into multiple implementation files, but do we ever split one implementation file into multiple header files? I'd prefer to not do the latter if possible.

hx235 commented:
> Is it a common pattern to put declarations and definitions into differently named files? MustHaveNotExisted() would be in expected_state.cc and expected_value.h according to this PR. I know we split db_impl.h into multiple implementation files, but do we ever split one implementation file into multiple header files? I'd prefer to not do the latter if possible.

Thanks!! Fixed.

jowlyzhang commented on db_stress_tool/expected_value.h:
```diff
@@ -0,0 +1,206 @@
+ public:
+  // Return whether value is expected not to exist from begining till the end
+  // of the read based on `pre_read_expected_value` and
+  // `pre_read_expected_value`.
```
nit: pre_read_expected_value \ post_read_expected_value. Similar nit for below `MustHaveNotExisted`.

In "value is expected not to exist ...", what does "value" refer to?

hx235 replied:
Good question ....

hx235 replied:
Fixed

jowlyzhang reviewed (commented):
Thanks for the PR. LGTM, just some minor comments.
